src/App/Application.cpp

```diff
@@ -2126,52 +2127,51 @@ void Application::initTypes()
 }
 
 namespace {
-pair<string, string> customSyntax(const string& s)
+
+pair<string, string> customSyntax(string_view strIn)
 {
+    if(strIn.size() < 2) {
+        return{};
+    }
+
+    char leadChr {strIn[0]};
+    string rest {strIn.substr(1)};
+
+    if (leadChr == '@') {
+        return {"response-file", rest};
+    }
+
+    if (leadChr != '-') {
+        return {};
+    }
+
 #if defined(FC_OS_MACOSX)
-    if (s.find("-psn_") == 0)
-        return make_pair(string("psn"), s.substr(5));
+    if (rest.find("psn_") == 0) {
+        return {"psn", rest.substr(4)};
+    }
 #endif
-    if (s.find("-display") == 0)
-        return make_pair(string("display"), string("null"));
-    else if (s.find("-style") == 0)
-        return make_pair(string("style"), string("null"));
-    else if (s.find("-graphicssystem") == 0)
-        return make_pair(string("graphicssystem"), string("null"));
-    else if (s.find("-widgetcount") == 0)
-        return make_pair(string("widgetcount"), string(""));
-    else if (s.find("-geometry") == 0)
-        return make_pair(string("geometry"), string("null"));
-    else if (s.find("-font") == 0)
-        return make_pair(string("font"), string("null"));
-    else if (s.find("-fn") == 0)
-        return make_pair(string("fn"), string("null"));
-    else if (s.find("-background") == 0)
-        return make_pair(string("background"), string("null"));
-    else if (s.find("-bg") == 0)
-        return make_pair(string("bg"), string("null"));
-    else if (s.find("-foreground") == 0)
-        return make_pair(string("foreground"), string("null"));
-    else if (s.find("-fg") == 0)
-        return make_pair(string("fg"), string("null"));
-    else if (s.find("-button") == 0)
-        return make_pair(string("button"), string("null"));
-    else if (s.find("-btn") == 0)
-        return make_pair(string("btn"), string("null"));
-    else if (s.find("-name") == 0)
-        return make_pair(string("name"), string("null"));
-    else if (s.find("-title") == 0)
-        return make_pair(string("title"), string("null"));
-    else if (s.find("-visual") == 0)
-        return make_pair(string("visual"), string("null"));
-//  else if (s.find("-ncols") == 0)
-//    return make_pair(string("ncols"), boost::program_options::value<int>(1));
-//  else if (s.find("-cmap") == 0)
-//    return make_pair(string("cmap"), string("null"));
-    else if ('@' == s[0])
-        return std::make_pair(string("response-file"), s.substr(1));
-    else
-        return make_pair(string(), string());
+
+    constexpr std::array knowns {"display",
+                                 "style",
+                                 "graphicssystem",
+                                 "widgetcount",

```

Comment:
widgetcount is a special parameter and its value is an empty string, not "null".